### DISCUSSION TITLE
Can't load any ggml models after Ooba Booga update

### DISCUSSION THREAD

**Discussion by githubuser1987** [oobabooga/text-generation-webui#2178] [Q&A]
Hi guys,

After updating the web-ui to use the new llama 1.50, none of the ggml models load anymore.
I get "ERROR:Could not find the quantized model in .pt or .safetensors format, exiting...".

---------------------------------
The following models are available:

1. anon8231489123_gpt4-x-alpaca-13b-native-4bit-128g
2. ggml-vic7b-uncensored-q5_1.bin
3. gozfarb_pygmalion-7b-4bit-128g-cuda
4. TheBloke_wizard-mega-13B-GGML
5. TheBloke_WizardLM-7B-uncensored-GPTQ
6. wizard-mega-13B-GPTQ-4bit-128g.no-act.order
7. wizard-mega-13B.ggml.q5_1.bin
8. WizardLM-7B-uncensored-GPTQ-4bit-128g.compat.no-act-order.safetensors
9. wizardLM-7B.ggml.q5_1.bin

Which one do you want to load? 1-9

2

INFO:Loading ggml-vic7b-uncensored-q5_1.bin...
ERROR:Could not find the quantized model in .pt or .safetensors format, exiting...

Done!
Press any key to continue . . .
-------------------------------------

I even redownloaded the OobaBooga and re-installed everything, but I still get the same error message.
Has anyone encountered that problem as well?

**Comment by m-spangenberg** [comment]
The llama.cpp update isn't backwards compatible with older GGML models.

**Reply by githubuser1987** [reply]
Thanks for the reply. Good to know.
However, even after downgrading to llama version 0.1.45 I still get the same error message.

**Reply by DavidInRacine** [reply]
> _The llama.cpp update isn't backwards compatible with older GGML models._

> **ERROR:The model could not be loaded because its checkpoint file in .bin/.pt/.safetensors format could not be located.**

All previously downloaded ggml models I tried failed, including the latest Nous-Hermes-13B-GGML model uploaded by _The Bloke_ five days ago, and downloaded by myself today.

I noticed a script in **text-generation-webui** folder titled **convert-to-safetensors.py** but I could not find any way to convert my older ggml .bin models which have not been updated yet on HuggingFace. The script descriptor says:

> _Converts a transformers model to safetensors format and shards it._
> 
> _This makes it faster to load (because of safetensors) and lowers its RAM usage while loading (because of sharding)._

https://github.com/oobabooga/text-generation-webui/blob/e471919e6d504e85ac1aa58ad6bf0d46d0d9323d/convert-to-safetensors.py#L3

All previously downloaded models were broken by the safetensors issue, and many models still on HuggingFace do not work. I wish there were something we could do about it (?) but this appears to require a re-quantization of every model... a huge effort by _The Bloke_ and others.

Life on the bleeding edge!

**Reply by DavidInRacine** [reply]
> Thanks for the reply. Good to know. However, even after downgrading to llama version 0.1.45 I still get the same error message.

Did you ever get ooba to work for you? I have been at this several hours today and no model works. All of them report these sort of messages with models uploaded to HuggingFace in the past 4-5 days:
```

2023-06-11 17:42:53 INFO:Loading Wizard-Vicuna-13B-Uncensored.ggmlv3.q5_1.bin...
2023-06-11 17:42:53 ERROR:The model could not be loaded because its checkpoint file in .bin/.pt/.safetensors format could not be located.
2023-06-11 17:43:08 ERROR:No model is loaded! Select one in the Model tab.
```

**Comment by viperwasp** [comment]
I don't understand to much about this. Like not backyards compatible etc. But I think this error can be solved by doing the following. 

1. Do not load these models using arguments. Use the default Start Windows or whatever it's called. Select the model via number in CMD launcher. 
2. If it still does not work. Load any model that does work. 
3. Then manually load the model from the internal settings that your having trouble with. If it's GGML make sure the Wbits and groupsize are blank. Make sure llama model is selected in drop down. Finally you may need to check or uncheck the setting that talks about loading with autogtp. Save settings, reload model. 

Do not use arguments always load from the selector. I don't think I can get GGML to work with arguments. 

I think that gets rid of my 
ERROR:The model could not be loaded because its checkpoint file in .bin/.pt/.safetensors format could not be located.

**Reply by DavidInRacine** [reply]
> I think that gets rid of my ERROR:The model could not be loaded because its checkpoint file in .bin/.pt/.safetensors format could not be located.

Thanks. I tried all that but nothing works. The only thing that works is, luckily, I kept a zipped archive of my **text-generation-webui** folder dated 05.14.2023. I know that some more recent ooba versions worked, but I didn't keep an archive of them.

Replacing the newer text-generation-webui folder with my old unzipped old one gets AI models working again, but of course oobabooga is frozen in time at May 14. If I update ooba, all models stop working. Nothing I tried makes them work.

Maybe it's time to switch to Kobold.cpp. I've been meaning to try it out.

**Reply by viperwasp** [reply]
Well I'm having my own form of issues just put up my own thread in the discussions. lol 
That is a shame I feel your pain. If I think of anything I will let you know. Weird question? 
Are you using the ooba that has the Windows update.bat or whatever it's called? 
Or are you using the former Ooba that really can only be updated via pip as far as I can tell? 
Just curious. 

The one I have the most success with is the one obtained from the 
[oobabooga-windows.zip](https://github.com/oobabooga/text-generation-webui/releases/download/installers/oobabooga_windows.zip) located under Windows One Click installer on found here https://github.com/oobabooga/text-generation-webui

I noticed if I went to the other one click installer section it's like a completely different thing and it's like bugged.

**Reply by DavidInRacine** [reply]
I am on Linux Mint

Despite many attempts and work-arounds, Ooba wouldn't install for me in KDE Neon  (another Debian distro) nor in Manjaro (Arch variant). Under both distros the ooba installer failed to compile the wheels. Only Linux Mint succeeded in building a working ooba folder, and of course that could all break when LM updates to "Victoria" later this month.

Open source AI feels like flying a 1909 biplane made of pigskin, nails and balsa wood.

**Reply by kujamara** [reply]
> Maybe it's time to switch to Kobold.cpp. I've been meaning to try it out.
> 
I feel your pain. None of my models will load in [oobabooga](https://github.com/oobabooga/text-generation-webui) (Windows) either, no matter what i try. [Kobold.cpp](https://github.com/LostRuins/koboldcpp) works great and can handle every model I've tried. Additionally, it takes up way less storage, it's my new go-to. [KoboldAI](https://github.com/henk717/KoboldAI) on the other hand, doesn't load models. So I'm sticking to Kobold Lite.

**Reply by viperwasp** [reply]
What CPU/GPU do you have Kujamara?  If you have GPU that comes in multiple varients like the 3060 can have 8GB or 12GB of VRAM tell us how much you have. And how much system ram?

**Reply by kujamara** [reply]
Thanks for your reply!

I am just playing around with my notebook, i was interested in how consumer devices without RTX and large amounts of RAM handle LLMs. Therefore I run all the models (7B or less, 2-Bit k-quant GGML) in CPU-only mode, because using my weak GPU actually increases the time/token.

Specs:
CPU: Intel i7-8550U (AVX2)
GPU: GeForce MX150 6GB
RAM: 8 GB

**Reply by DavidInRacine** [reply]
I have 64 GB RAM and AMD Ryzen 7 Pro 5850U CPU — 8 CPU Cores, 16 Threads, 4.4GHz max, 1.9 GHz base clock, 16MB L3 Cache. It handles Wizard-Vicuna and other 13B models at what I consider a reasonable pace, say 1-2 minutes for a full paragraph following the initial model load. The first paragraph might take 3-4 minutes. Overall it's fast enough to experiment with and learn some things, but not enough to serve as an AI production machine.

**Reply by viperwasp** [reply]
Yes you likely are limited to 7B or less. Your problem with "Ooba" might be that I don't think it can run 2-Bit and or K-quant yet. It can run 7B GGML 4_0, 4_1, 5_0,5_0 and at least a few others support for K-quant I think is underway. However unless i missed an update which is possible that is likely your issue.

**Reply by kujamara** [reply]
I also tried this [alpaca-native](https://huggingface.co/Sosaka/Alpaca-native-4bit-ggml) version, didn't work on ooga. But it runs with alpaca.cpp and as mentioned before with koboldcpp. 

Then I tried using [lollms-webui](https://github.com/ParisNeo/lollms-webui) and [alpaca-electron](https://github.com/ItsPi3141/alpaca-electron). Didn't work neither with old ggml nor with k quant ggml.

Yes, I hope the ooga team will add the compatibility with 2-bit k quant ggml models soon. Ooga is just the best looking and most versatile webui imo and i am definitely gonna use it if it's working, but im fine with koboldcpp for now.

Also I have to add that i run everything on my external SSD. I don't know if that is causing problems with some of these.

**Reply by ossirytk** [reply]
It'll be interesting to see the gpu only ggml changes too. I can just about run some 13B models in my 8gb of vram now. The performance is pretty good. Would like to see that in the text ui.

**Reply by DavidInRacine** [reply]
I downloaded oobabooga this morning – yesterday's updates – and discarded my previous ooba folder, then recompiled it in Linux (KDE Neon rolling OS).

All problems appear to be resolved... no wheels problems, no safetensors problems.  I am able to load older ggml models as well as the new K-method ggml models. (I don't have NVIDIA to test the gptq models.)

I will try to replicate this experience later today in Linux Mint and in Manjaro KDE (Arch) distros.

Hooray for oobabooga users! We're back into AI model testing with the latest ooba!

**Reply by DavidInRacine** [reply]
I am confirming success compiling and running oobabooga today in all three of my Linux distros, including: Linux Mint 21.1 (stable), KDE Neon (rolling plasma), and Manjaro KDE (rolling Arch). There are no more wheels problems and no more safetensors problems. All 13B models I tried ran perfectly.

Well done!

**Reply by kujamara** [reply]
Great, then I will also try it again! Thanks for letting me know.

**Comment by DavidInRacine** [comment]
> > Maybe it's time to switch to Kobold.cpp. I've been meaning to try it out.
> 
> I feel your pain. None of my models will load in [oobabooga](https://github.com/oobabooga/text-generation-webui) (Windows) either, no matter what i try. [Kobold.cpp](https://github.com/LostRuins/koboldcpp) works great and can handle every model I've tried. Additionally, it takes up way less storage, it's my new go-to. [KoboldAI](https://github.com/henk717/KoboldAI) on the other hand, doesn't load models. So I'm sticking to Kobold Lite.

I only dabbled with Kobold briefly a couple nights ago. I did not realize it doesn't load models. Great info but 😢.

Maybe I'll try Kobold Lite. I couldn't get Kobold to run in KDE Neon (an Ubuntu-Debian derived rolling distro) but perhaps Kobold Lite will run there?

**Reply by kujamara** [reply]
You definitely have to try, good luck!

**Reply by DavidInRacine** [reply]
I am hopeful an open source LangChain project will become viable for those of us running CPU-only, but so far, from what I can see, it's mostly just talk and future possibilities.
